src/rpc/claimtrie.cpp

```diff
@@ -183,11 +183,14 @@ UniValue getvalueforname(const UniValue& params, bool fHelp)
     std::string sValue;
     if (!getValueForClaim(claim.outPoint, sValue))
         return ret;
+
     ret.push_back(Pair("value", sValue));
     ret.push_back(Pair("txid", claim.outPoint.hash.GetHex()));
     ret.push_back(Pair("n", (int)claim.outPoint.n));
-    ret.push_back(Pair("amount", claim.nAmount));
-    ret.push_back(Pair("effective amount", claim.nEffectiveAmount)); 
+    ret.push_back(Pair("amount", ValueFromAmount(claim.nAmount)));
+    claimsForNameType claimsForName = pclaimTrie->getClaimsForName(name);
```

Comment:
The effective amount is only being calculated for one claim here.

As for a way to calculate the effective amounts for multiple claims, that functionality is already in place in the only places I know of that it needs to be.